emeritus-tech/landing-pages-libraries
# [CPT-471] Create stylesheets and markup template for CDS & CDM landing pages.

🎟 [**Jira Ticket**](https://emeritus.atlassian.net/browseCPT-471)



### 🚢 Deployment Instructions

* N/A

### ✅ QA instructions

Smoke test the whole index page taking into account the following:

* To test Columbia Digital Strategies
  * In `src/index.scss` replace the only `@import` with the following address: `@import "./modules/columbia/digital_strategies/index.scss"`.
  * Run command `npm run build` to compile.
  * Open `dist/index.html` on browser:
    * Design should look exactly as original CDM [landing page](https://admissions.emeritus.org/programs/columbia-business-school-executive-education/digital-strategies)
    * Modals should work as expected (without the `modals.js` file will only function in fallback mode, meaning that only clicking the `X` button will close it and `ESC` key or clicking outside will not)
    * Testimonial slider should work as expected using `CSS` only (not entirely the same as the original landing page one)
    * Load more functionality `CSS` only should work as expected

* To test Columbia Digital Marketing
  * In `src/index.scss` replace the only `@import` with the following address: `@import "./modules/columbia/digital_marketing/index.scss"`
  * Run command `npm run build` to compile.
  * Open `dist/index.html` on browser:
    * Design should look exactly as original CDM [landing page](https://admissions.emeritus.org/programs/columbia-business-school-executive-education/digital-marketing#download-brochure)
    * Modals should work as expected (without the `modals.js` file will only function in fallback mode, meaning that only clicking the `X` button will close it and `ESC` key or clicking outside will not)
    * Testimonial slider should work as expected using `CSS` only (not entirely the same as the original landing page one)
    * Responsiveness should work as expected

### ⚠️ Risks

* Low

### 🖼 Previews

* #### Columbia Digital Strategies

Desktop
![screencapture-file-Users-bocha-Code-rootstrap-landing-pages-libraries-styles-dist-index-html-2020-04-23-15_02_34](https://user-images.githubusercontent.com/251192/80136528-33cf1300-8578-11ea-98d9-ad0c8d45a714.jpg)

[Mobile](https://user-images.githubusercontent.com/251192/80136503-2e71c880-8578-11ea-8419-0818d86e8812.jpg)


* #### Columbia Digital Marketing

Desktop
![screencapture-file-Users-bocha-Code-rootstrap-landing-pages-libraries-styles-dist-index-html-2020-04-23-15_10_05](https://user-images.githubusercontent.com/251192/80136248-d76bf380-8577-11ea-994e-37bfd18e8f12.jpg)

[Mobile](https://user-images.githubusercontent.com/251192/80136234-d20ea900-8577-11ea-84f7-9375115b9607.jpg)


### 📝 Notes

* As requested, hero form and modal form wasn't implemented for now.
* Sticky nav will not show up (will not be broken tho) until the following new refactored `JS` file is implemented: [`sticky.js`](https://gist.github.com/bocha/cc59f3b132273af0164fd7685e8a7978)
* Modals will not work entirely as expected unless original `modals.js` file is implemented.
